### PR TITLE
chore(tracing): Improve tool for formatting traces

### PR DESCRIPTION
**Pull request by Aristotelis2002**
Rewrote the tool in Rust instead of bash so it integrates better with the rest of our system. Also, it now prints arrays on multiple lines.

More information in README.md.

### CODE REVIEW

File: testing/format_trace/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[package]
```
[the remark this answers is not in the document]
Comment (Xearty) [reply testing/format_trace/Cargo.toml:1]: `[workspace]` implies there will be multiple crates in `format_trace`. If this is not the case then in my opinion it's better to leave it as `[package]` and not make it part of a workspace.

[the remark this answers is not in the document]
Comment (Aristotelis2002) [reply testing/format_trace/Cargo.toml:1]: Perhaps you meant adding it to the main Cargo.toml workspace members which I did in my last change.

File: testing/format_trace/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    let src_filename = args[1].clone();
+    let dest_filename = args[2].clone();
+    let mut ser_json: Value = serialize_file(src_filename);
+
+    if let Some(array) = ser_json.as_array_mut() {
+        for object in array {
+            sort_json_fields(object);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Aristotelis2002) [reply testing/format_trace/src/main.rs]: Done.

File: testing/format_trace/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (smanilov) [reply testing/format_trace/Cargo.toml:1]: I think I actually wanted a workspace here. Let me confirm and I'll get back to you.

**Pull request merged**